app/models/job.ts

```diff
@@ -428,6 +442,51 @@ export class Job extends Record {
     return data.length !== 0;
   }
 
+  /**
+   * Determine if the current job is the same as an already running job. If so, replace the
+   * current job's requestId with that of the running job's. Add an attachedStatus property for
+   * later referencing.
+   *
+   * @param transaction - The transaction to use for saving the job
+   * @throws {@link Error} if the job is invalid
+   */
+  async maybeAttach(transaction: Transaction): Promise<void> {
+    const rows = await transaction('jobs')
+      .select('requestId', 'shapeFileUrl')
+      .where({ username: this.username, request: this.request })
+      .andWhere('status', JobStatus.ACCEPTED)
+      .orWhere('status', JobStatus.RUNNING);
+    if (Array.isArray(rows) && rows.length > 0) {
+      const originalId = this.requestId;
+      // if a shapefile is not present then database query returned all matching running requests
+      if (this.shapeFileUrl === '') {
+        const assumedId = rows.shift()['requestId'];
+        this.attachedStatus = { didAttach: true, originalId, assumedId };
+      // otherwise, compare checksum of current shapefile to other matching requests shapefile
+      } else {
+        const store = defaultObjectStore();
+        const thisSum = await objectStoreChecksum(store, this.shapeFileUrl);
+        let otherSum = '';
```

Comment:
I don't think this needs to be declared here, it can be a `const` in your `for` loop.

Reply:
It looks like `const`'s and `let`'s are block scoped so declaring outside of the `for` is needed here because the variable is used after that block. Also explains why a `const` can be used inside of a `for` loop: a variable declared this way falls out of scope and doesn't violate `const`'s prevention of variable reassignment.

Reply:
It doesn't look like `otherSum` is used outside the `for` loop, unless I'm misreading something.